src/operator/nn/activation.cc

```diff
@@ -27,10 +27,10 @@
 #include "./activation-inl.h"
 #include "../mshadow_op.h"
 #include "../tensor/elemwise_unary_op.h"
-#if MXNET_USE_MKLDNN == 1
+#if MXNET_USE_CUDNN == 0 && MXNET_USE_MKLDNN == 1
```

Comment:
If USE_CUDNN=1, then MKL-DNN code in this file will not be compiled?

Reply:
If MXNet is built with CUDNN and MKL-DNN enable at the same time, I hope behavior is the activation will be dispatched to CUDNN if model context is gpu, otherwise it should be dispatched to MKL-DNN implementation. But with this change, MKL-DNN implementation will not be compiled.

Reply:
This is a good point, the latest commit is just a test. What I observed is that we are calculating the backward in MKLDNN even if we have CUDNN.
via ActivationGradComputeExCPU as per:

https://github.com/apache/incubator-mxnet/blob/master/src/operator/nn/activation.cc#L189

Reply:
I‘m not sure. But I think if MKL-DNN is enabled, the execution should be dispatched to FComputeEx according to the infer storage, but seems FComputeEx is not registered for gpu Activation. Need inputs from expert of execution engine.

Reply:
Seems FComputeEx takes precedence:

https://github.com/apache/incubator-mxnet/blob/master/src/imperative/imperative.cc#L65


@eric-haibin-lin  @piiswrong

Reply:
My commit is passing all the unit tests, as it doesn't use MKLDNN if CUDNN is enabled. I think is the right way to go as of now, since we can't selectively dispatch to MKLDNN depending on the context without a bigger refactor.